src/fifo/FifoControllerBase.cpp

```diff
@@ -30,33 +30,34 @@ FifoControllerBase::FifoControllerBase(uint32_t totalFrames, uint32_t threshold)
 {
 }
 
-int32_t FifoControllerBase::getFullFramesAvailable() const {
-    return static_cast<int32_t>(getWriteCounter() - getReadCounter());
+uint32_t FifoControllerBase::getFullFramesAvailable() const {
```

Comment:
This value can go negative if there is an underflow! Please do not convert it to signed.